### DISCUSSION TITLE
How to define username/password for clab config command?

### DISCUSSION THREAD

**Discussion by riddler63** [srl-labs/containerlab#1474] [Q&A]
Hello,

I'm trying to use clab config, but cannot find a way to define username/password for ssh. 
What I am missing here? 
I have not found the examples with username/password in topology file.

sudo containerlab config -t ipv6_only.clab.yml -f leaf1 -p ./cfg-templates -l fullcfg-leaf
INFO[0000] Parsing & checking topology file: ipv6_only.clab.yml 
WARN[0000] leaf1: **SSH credentials for node leaf1 of type ceos not found, cannot configure**

**Comment by hellt** [comment]
@kellerza I think this has been hard coded, afair?

**Reply by kellerza** [reply]
It is pulled from the node registry today.

In labctl you can overwrite these with variables - see [here](https://labctl.net/guide/config-engine-tx#ssh-settings).

We should really have a chat when you are back from leave on how to move some of the labctl Tx (incl scrapligo) code into containerlab

**Comment by steiler** [comment]
The credentials registered in the noderegistry are being used here.
However for the CEOS instance there is no credentials provided
https://github.com/srl-labs/containerlab/blob/main/nodes/ceos/ceos.go#L55

**Reply by riddler63** [reply]
Hello @steiler , 

Then it's bug? Should I create an issue? 

Or maybe it is better to create a feature request to allow ssh_username/ssh_password per node, per kind for clab config?

**Reply by steiler** [reply]
This PR contains the default ceos credentials.
https://github.com/srl-labs/containerlab/pull/1476
Try the resulting binary. Here is the link to the cicd generated binary artifact: 
https://github.com/srl-labs/containerlab/suites/14463609303/artifacts/816913501
Let us know if this works.

**Reply by riddler63** [reply]
Hello,

It seems that the cEOS platform is not fully supported by clab config. 
The previous message have gone, but I've got the following warning.

```
sudo containerlab config -t ipv6_only.clab.yml -f leaf1 -p ./cfg-templates -l fullcfg-leaf
INFO[0000] Parsing & checking topology file: ipv6_only.clab.yml 
WARN[0000] leaf1: no transport implemented for kind: ceos
```

**Reply by kellerza** [reply]
@riddler63 I implemented the current transport mechanism before scapligo was around. 

That is why I recommended you try labctl. It uses the clab config engine, adds a UI for config engine & the transport mechanism now uses scrapligo

The previous PR to add scrapligo was never merged and at some point I'll merge some of my work in labctl into containerlab. It's just the config engine user interface that will be unique to labctl
